src/components/molecules/Jukebox/Jukebox.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import classNames from "classnames";
+
+import { CHAT_MESSAGE_TIMEOUT, YOUTUBE_SHORT_URL_STRING } from "settings";
+
+import { User } from "types/User";
+
+import { WithId } from "utils/id";
+import { getYoutubeEmbedFromUrl, isStringAValidUrl } from "utils/url";
+
+import { useJukeboxChat } from "hooks/jukebox";
+import { useUser } from "hooks/useUser";
+import { useVenueId } from "hooks/useVenueId";
+
+import { ButtonNG } from "components/atoms/ButtonNG";
+import { InputField } from "components/atoms/InputField";
+
+import "./Jukebox.scss";
+
+type JukeboxTypeProps = {
+  recentVenueUsers: readonly WithId<User>[];
+  updateIframeUrl: Dispatch<SetStateAction<string>>;
+};
+const Jukebox: React.FC<JukeboxTypeProps> = ({
+  recentVenueUsers,
+  updateIframeUrl,
+}) => {
+  const { register, handleSubmit, watch, reset } = useForm<{
+    jukeboxMessage: string;
+  }>({
+    mode: "onSubmit",
+  });
+  const [isSendingMessage, setMessageSending] = useState(false);
+  const chatValue = watch("jukeboxMessage");
+  const venueId = useVenueId();
+  const { userId } = useUser();
+  const [filteredUser] =
+    recentVenueUsers.filter((vUser) => vUser.id === userId) || [];
+  const tableName = venueId ? filteredUser?.data?.[venueId].table ?? "" : "";
+
+  const { sendJukeboxMsg, messagesToDisplay } = useJukeboxChat({
+    venueId,
+    tableId: tableName,
+  });
+
+  const filteredMessages = messagesToDisplay.filter(
+    (msg) => msg.tableId === tableName
+  );
+
+  useEffect(() => {
+    if (messagesToDisplay.length) {
+      const [lastMessage] = messagesToDisplay;
+      if (lastMessage?.text?.includes(YOUTUBE_SHORT_URL_STRING)) {
+        const youtubeUrl = getYoutubeEmbedFromUrl(lastMessage?.text);
+        updateIframeUrl(youtubeUrl);
+        return;
+      }
+      if (isStringAValidUrl(lastMessage?.text)) {
+        updateIframeUrl(lastMessage.text);
+        return;
+      }
+      return;
+    }
+  }, [messagesToDisplay, updateIframeUrl]);
+
+  const sendMessageToChat = handleSubmit(({ jukeboxMessage }) => {
+    const url = jukeboxMessage;
+    setMessageSending(true);
+    sendJukeboxMsg({ message: jukeboxMessage, url });
+    reset();
+  });
+
+  // This logic disallows users to spam into the chat. There should be a delay, between each message
```

Comment:
The spam-prevention code here looks like it could use the concept of debouncing. Useful article: [link](https://dev.to/heyitsarpit/improve-performance-of-your-react-app-with-debouncing-build-your-own-usedebounce-hook-1fb1).

We already have a `useDebounce` hook in react-use lib but it may be it somewhat limiting, so maybe [this](https://www.npmjs.com/package/use-debounce) one suits better.

Reply:
probably gonna call it OOS, so that we can implement debounce to both chat/jukebox